pandas/tests/indexing/test_loc.py

```diff
@@ -2971,6 +2971,28 @@ def test_loc_periodindex_3_levels():
     assert mi_series.loc[(p_index[0], "A", "B")] == 1.0
 
 
+@pytest.mark.parametrize("utc", [False, True])
+def test_loc_datetime_assignment_dtype_does_not_change(utc):
+    # GH#49837
+    df = DataFrame(
+        {
+            # Checking with and without UTC as the original bug didn't occur
+            # for utc=True
+            "date": to_datetime(
+                [datetime(2022, 1, 20), datetime(2022, 1, 22)], utc=utc
+            ),
+            "update": [True, False],
+        }
+    )
+    # We don't expect the dataframe to be changes at all
```

Comment:
Can you remove the comments?  In general, we try to keep them out of tests

Reply:
NP.
Any special reason or just conventions?

Reply:
Tests should be simply enough to be self-explanatory. Additionally, the issue ref provides the context